code/presentation/mb-client/src/screens/home/find-masjid.tsx

```diff
@@ -0,0 +1,234 @@
+import { useNavigation } from '@react-navigation/native';
+import * as Location from 'expo-location';
+import React, { useState } from 'react';
+import {
+  Button,
+  FlatList,
+  StyleSheet,
+  Text,
+  TextInput,
+  TouchableOpacity,
+  View,
+} from 'react-native';
+import type { Details, Region } from 'react-native-maps';
+import MapView, { enableLatestRenderer, Marker } from 'react-native-maps';
+
+import type { MasjidWithDistance, NamazTimings } from '@/api/masjid/types';
+import { useMasjids } from '@/api/masjid/use-masjids';
+import { getNextNamaz } from '@/api/masjid/util';
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+  },
+  map: {
+    flex: 1,
+  },
+  mapContainer: {
+    flex: 1,
+  },
+  input: {
+    borderWidth: 1,
+    borderColor: 'gray',
+    borderRadius: 5,
+    padding: 10,
+    margin: 10,
+  },
+  buttonContainer: {
+    alignItems: 'center',
+    marginTop: 10,
+  },
+  masjidDetailsContainer: {
+    backgroundColor: '#f0f0f0',
+    padding: 10,
+    marginTop: 10,
+    borderRadius: 5,
+    borderWidth: 1,
+    borderColor: '#ccc',
+  },
+  masjidName: {
+    fontSize: 18,
+    fontWeight: 'bold',
+    marginBottom: 5,
+    color: '#333',
+  },
+  masjidDistance: {
+    fontSize: 16,
+    marginBottom: 5,
+    color: '#555',
+  },
+  masjidAddress: {
+    fontSize: 16,
+    color: '#666',
+  },
+  listItem: {
+    padding: 10,
+    borderBottomWidth: 1,
+    borderBottomColor: '#ccc',
+  },
+  listItemText: {
+    fontSize: 16,
+  },
+});
```

Comment:
can we use native wind instead? style should be used only when necessary

Reply:
Done